proto/process/agent.proto

```diff
@@ -570,35 +578,134 @@ message ServiceSessionAffinityConfig {
     int32 clientIPTimeoutSeconds = 1;
 }
 
+// reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/core/v1/generated.proto#L2107
+message Node {
+	Metadata metadata = 1;
+	NodeSpec spec = 2;
+	NodeStatus status = 3;
+	bytes yaml = 4;
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/core/v1/generated.proto#L2312
+message NodeSpec {
+	string podCIDR = 1;
+	repeated string podCIDRs = 3;
+	string providerID = 2;
+	bool unschedulable = 4;
+	repeated Taint taints = 5;
+	ConfigMapNodeConfigSource configSource = 6;
+
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/core/v1/generated.proto#L515
+message ConfigMapNodeConfigSource {
+	string namespace = 1;
+	string name = 2;
+	string uid = 3;
+	string resourceVersion = 4;
+	string kubeletConfigKey = 5;
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/core/v1/generated.proto#L2348
+// skipping NodeConfigStatus for now.
+message NodeStatus {
+	// Key/Value of a resource type and a corresponding quantity as an int64.
+	map<string, int64> capacity = 1;
+	// Key/Value of a resource type and a corresponding quantity as an int64.
+	map<string, int64> allocatable = 2;
+	repeated NodeAddress addresses = 3;
+	repeated NodeCondition conditions = 4;
+	NodeDaemonEndpoints daemonEndpoints = 5;
+	NodeSystemInfo nodeInfo = 6;
+	repeated string volumesInUse = 7;
+	repeated AttachedVolume volumesAttached = 8;
+	repeated ContainerImage images = 9;
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/core/v1/generated.proto#L2158
+message NodeCondition {
+	string type = 1;
+	string status = 2;
+	int64 lastHeartbeatTime = 3;
+	int64 lastTransitionTime = 4;
+ 	string reason = 5;
+ 	string message = 6;
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/core/v1/generated.proto#L777
+message ContainerImage {
+	repeated string names = 1;
+	int64 sizeBytes = 2;
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/core/v1/generated.proto#L2127
+message NodeAddress {
+	string type = 1;
+	string address = 2;
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/core/v1/generated.proto#L2240
+message NodeDaemonEndpoints {
+	int32 kubeletEndpoint = 1;
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/core/v1/generated.proto#L82
+message AttachedVolume {
```

Comment:
This one i agree we should probably flatten as well, either in the agent or in the writer.

Reply:
removed it, as  discussed offline.